src/core/exec-invoke.c

```diff
@@ -5845,8 +5857,33 @@ int exec_invoke(
                         _cleanup_free_ char *jb = strv_join(bad_variables, ", ");
                         log_warning("Invalid environment variable name evaluates to an empty string: %s", strna(jb));
                 }
-        } else
-                final_argv = command->argv;
+        }
+
+        if (FLAGS_SET(command->flags, EXEC_COMMAND_PREFIX_SHELL)) {
+                r = strv_extendf(&prefixed_argv, "%s%s", command->argv[0][0] == '-' ? "-" : "", path);
```

Comment:
why use format string here? you can just use strv_extend() here

Reply:
hmm? I don't follow? We need to concat two strings and push the result to strv?

Reply:
right, ignore me.